### ISSUE TITLE
a11y: wrong header element for group title

### ISSUE THREAD

**Issue by Danielle254** [LetsGetTechnical/elecretanta#531]
'/dashboard' page (after at least 1 group has been added)

Group title should be an `<h3>` element, not an `<h2>` to maintain proper semantic HTML hierarchy.

"Dashboard" - h1
"My Groups" - h2
"Group Title" - h3

![Image](https://github.com/user-attachments/assets/e72d825a-3f71-4ed1-aa28-4756420a7793)
